### PR TITLE
Incorporation of Expansion filters into the MGXS Class

### PR DESCRIPTION
**Pull request by nelsonag**
This PR completes the open item from #990, using the Legendre expansion filters.  Therefore, `scatter-n`, `scatter-pn`, `scatter-yn`, the `nu-` equivalents, `flux-yn`, and `total-yn` scores are removed, but a deprecation error is left in their place.

All notebooks, docs and tests have been checked/updated.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -99,4 +99,7 @@ examples/jupyter/plots
 .tox/
 .python-version
 .coverage
-htmlcov
\ No newline at end of file
+htmlcov
+
+# Test data
+tests/xsdir
```
[the remark this answers is not in the document]
Comment (nelsonag) [reply .gitignore]: - [x]

File: docs/source/usersguide/tallies.rst [hunk 2]
```diff
@@ -21,7 +21,12 @@ region of phase space, as in:
 Thus, to specify a tally, we need to specify what regions of phase space should
 be included when deciding whether to score an event as well as what the scoring
 function (:math:`f` in the above equation) should be used. The regions of phase
-space are called *filters* and the scoring functions are simply called *scores*.
+space are generally called *filters* and the scoring functions are simply
+called *scores*.
+
+The only cases when *filters* do not correspond directly with the regions of
```
[the remark this answers is not in the document]
Comment (nelsonag) [reply docs/source/usersguide/tallies.rst]: - [x]

File: openmc/mgxs/mgxs.py [hunk 3]
```diff
@@ -1923,6 +1923,10 @@ def get_pandas_dataframe(self, groups='all', nuclides='all',
             if 'group out' in df:
                 df = df[df['group out'].isin(groups)]
 
+        # Add the Legendre bin to the column if it exists
+        if 'legendre' in df:
+            columns += ['legendre']
```
[the remark this answers is not in the document]
Comment (nelsonag) [reply openmc/mgxs/mgxs.py]: - [x]

File: src/input_xml.F90 [hunk 4]
```diff
@@ -2541,107 +2535,22 @@ subroutine read_tallies_xml()
...
-              if (imomstr >= YN_LOC) then
-                n_bins = (n_order + 1)**2
-              else
-                n_bins = n_order + 1
-              end if
-              ! We subtract one since n_words already included
-              n_new = n_new + n_bins - 1
-              exit
-            end if
-          end do
```
[the remark this answers is not in the document]
Comment (nelsonag) [reply src/input_xml.F90]: It was quite rewarding...

File: openmc/tallies.py [hunk 5]
```diff
@@ -388,6 +386,14 @@ def scores(self, scores):
 
             # If score is a string, strip whitespace
             if isinstance(score, str):
+                # Check to see if scores are deprecated before storing
+                for deprecated in ['scatter-', 'nu-scatter-', 'scatter-p',
+                                   'nu-scatter-p', 'scatter-y', 'nu-scatter-y',
+                                    'flux-y', 'total-y']:
+                    if score.startswith(deprecated):
+                        msg = score.strip() + ' is deprecated and should no ' \
```
[the remark this answers is not in the document]
Comment (nelsonag) [reply openmc/tallies.py]: - [x]

File: src/summary.F90 [hunk 6]
```diff
@@ -399,24 +452,69 @@ subroutine write_materials(file_id)
...
-        else
-          nucnames(j) = nuclides_MG(m%nuclide(j))%obj%name
+      if (run_CE) then
+        num_nuclides = m % n_nuclides
+        num_macros = 0
+      else
+        ! Find the number of macroscopic and nuclide data in this material
+        num_nuclides  = 0
+        num_macros = 0
+        k = 1
```
[the remark this answers is not in the document]
Comment (nelsonag) [reply src/summary.F90]: - [x] (and neither is the line below it; `n = 1`)

File: src/tallies/tally.F90 [hunk 7]
```diff
@@ -2101,98 +2006,10 @@ subroutine expand_and_score(p, t, score_index, filter_index, score_bin, &
...
-             t % results(RESULT_VALUE, score_index, filter_index) &
-             + score * calc_pn(n, p % mu)
-      end do
-      i = i + t % moment_order(i)
-
-
-    case default
 !$omp atomic
       t % results(RESULT_VALUE, score_index, filter_index) = &
            t % results(RESULT_VALUE, score_index, filter_index) + score
```
[the remark this answers is not in the document]
Comment (nelsonag) [reply src/tallies/tally.F90]: - [x]

File: tests/regression_tests/track_output/test.py [hunk 8]
```diff
@@ -19,7 +19,7 @@ def _test_output_created(self):
     def _get_results(self):
         """Digest info in the statepoint and return as a string."""
         # Run the track-to-vtk conversion script.
-        call(['../../scripts/openmc-track-to-vtk', '-o', 'poly'] +
+        call(['../../../scripts/openmc-track-to-vtk', '-o', 'poly'] +
```
[the remark this answers is not in the document]
Comment (nelsonag) [reply tests/regression_tests/track_output/test.py]: Not a problem

File: tests/regression_tests/tallies/test.py [hunk 9]
```diff
@@ -17,53 +18,53 @@ def test_tallies():
 
     azimuthal_bins = (-3.14159, -1.8850, -0.6283, 0.6283, 1.8850, 3.14159)
     azimuthal_filter = AzimuthalFilter(azimuthal_bins)
-    azimuthal_tally1 = Tally()
+    azimuthal_tally1 = Tally(tally_id=1)
```
[the remark this answers is not in the document]
Comment (nelsonag) [reply tests/regression_tests/tallies/test.py]: It was an intermediate debug step that actually isnt needed anymore

**Comment by nelsonag** [comment]
@paulromano I added the tests on my more_mgxs_tests branch.  I am working on incorporating in to this branch.  A few things: no matter what the test inputs will not match since the scores, filters, and filter indices have changed.  Next, I'm dealing with two issues.  The first I am perplexed about, the second I apparently missed a case when I tested and will need a moment to fix.

The first is that the histogram cases have a different standard deviation after this PR than they do with the code in develop.  The means are the same.  I have no thoughts on this just yet.

**Comment by paulromano** [comment]
Ok. Let me know when you figure out the histogram issue; for now I'll put the "On hold" label.

**Comment by nelsonag** [comment]
Ok just some notes:
- Tests have been implemented before putting the API changes in
- The issue with the correction has been fixed.
- The standard deviation errors still persist.  They are present for these cases:
  - In the new mgxs_library_histogram: all "consistent nu-scatter matrix" types [with histogram bins] have their standard deviations slightly different.
  - In the new mgxs_library_correction: "scatter matrix" and "nu-scatter matrix" types [legendre] for the test problem's fuel and water materials [but not the Zr!?] have their standard deviations slightly different.
  - I will figure out the above and let you know when this PR is ready to move on.

**Comment by nelsonag** [comment]
Ok just some notes:
- Tests have been implemented before putting the API changes in.
- The issue with the correction code has been fixed.
- The standard deviation differences still persist, but they are for known reasons!
  - The reason was obvious after I thought about it.  This PR changes the score types/filters used to generate the data to do the above work.  For example, a "consistent scatter matrix" MGXS type used to have a "scatter-0" matrix and a "scatter-1" outgoing energy vector.  Now, because we dont have the ability to create "scatter-1", it seems more economical to just have the "scatter-0" matrix be a "scatter" matrix with a P1 filter.  Then, I calculate the equivalent scatter-1 outgoing energy vector from that matrix.  Doing that step inherently produces a different standard deviation.  I confirmed with hand calcs.

I'm about to update the tests to accept the new standard deviations and push to a new branch since its the simplest way to have the commit history be re-ordered to have the new tests in place before these changes.  That will make me need to do a new PR.

**Pull request closed without merging**